src/Draco.Compiler/Internal/Symbols/Metadata/MetadataSymbol.cs

```diff
@@ -108,4 +109,71 @@ internal static class MetadataSymbol
     private static FunctionSymbol SynthetizeConstructor(
         MetadataTypeSymbol type,
         MethodDefinition ctorMethod) => new SynthetizedMetadataConstructorSymbol(type, ctorMethod);
+
+    /// <summary>
+    /// Gets documentation for given <paramref name="symbol"/>.
+    /// </summary>
+    /// <param name="symbol">The <see cref="Symbol"/> to get documentation for.</param>
+    /// <returns>The documentation, or empty string, if no documentation was found.</returns>
+    public static string GetDocumentation(Symbol symbol)
+    {
+        var assembly = symbol.AncestorChain.OfType<MetadataAssemblySymbol>().FirstOrDefault();
+        if (assembly is null) return string.Empty;
+        var documentationName = GetPrefixedDocumentationName(symbol);
+        var root = assembly.AssemblyDocumentation?.DocumentElement;
+        var xml = root?.SelectSingleNode($"//member[@name='{documentationName}']")?.InnerXml ?? string.Empty;
+        return string.Join(Environment.NewLine, xml.ReplaceLineEndings("\n").Split('\n').Select(x => x.TrimStart()));
+    }
+
+    /// <summary>
+    /// Gets the full name of a <paramref name="symbol"/> used to retrieve documentation from metadata.
+    /// </summary>
+    /// <param name="symbol">The symbol to get documentation name of.</param>
+    /// <returns>The documentation name, or empty string, if <paramref name="symbol"/> is null.</returns>
+    public static string GetDocumentationName(Symbol? symbol) => symbol switch
+    {
+        FunctionSymbol function => GetFunctionDocumentationName(function),
+        TypeParameterSymbol typeParam => GetTypeParameterDocumentationName(typeParam),
+        null => string.Empty,
+        _ => symbol.MetadataFullName,
+    };
+
+    /// <summary>
+    /// The documentation name of <paramref name="symbol"/> with prepended documentation prefix.
+    /// </summary>
+    /// <param name="symbol">The symbol to get prefixed documentation name of.</param>
+    /// <returns>The prefixed documentation name, or empty string, if <paramref name="symbol"/> is null..</returns>
+    public static string GetPrefixedDocumentationName(Symbol? symbol) => $"{symbol switch
+    {
+        TypeSymbol => "T:",
+        ModuleSymbol => "T:",
+        FunctionSymbol => "M:",
+        PropertySymbol => "P:",
+        FieldSymbol => "F:",
+        _ => string.Empty,
+    }}{GetDocumentationName(symbol)}";
```

Comment:
You could extract a `GetDocumentationPrefix` function to avoid doing a switch in interpolation